// Copyright 2020 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

import "google/protobuf/any.proto";

package aistreams;

// This is the set of all known pre-defined types.
enum PacketTypeId {
  PACKET_TYPE_UNKNOWN = 0;
  PACKET_TYPE_JPEG = 1;
  PACKET_TYPE_RAW_IMAGE = 2;
  PACKET_TYPE_PROTOBUF = 3;
  PACKET_TYPE_STRING = 4;
  PACKET_TYPE_GSTREAMER_BUFFER = 5;
  PACKET_TYPE_CONTROL_SIGNAL = 6;
}

// The message that represents the data type of a packet.
message PacketType {
  // The type id.
  PacketTypeId type_id = 1;

  // This is the human readable representation of the type.
  //
  // This is not required to be populated; use the type_id field
  // above for identification purposes.
  string type_name = 2;

  // Additional information to fully specify the type, if any.
  google.protobuf.Any type_descriptor = 3;
}
